students/Daniel_Rodriguez/Lesson02/Assignment/charges_calc.py

```diff
@@ -0,0 +1,161 @@
+
+    parser.add_argument('-i', '--input', help='Input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='Output JSON file',
+                        required=True)
+
+    parser.add_argument('-d', '--debug', help='Debug message level:\n'
+                                              '0: No debug messages or '
+                                              'log file\n'
+                                              '1: Only error messages\n'
+                                              '2: Error messages and warnings\n'
+                                              '3: Error messages, warnings,'
+                                              'and debug messages',
+                        required=False)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    load_rentals_file
+    :param filename:
+    :return:
+    """
+    logging.debug('Reading input JSON file')
+    logging.debug('Inside load_rentals_file function')
+    logging.debug('Opening Filename: {}'.format(filename))
+
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+            logging.debug('Reading data file...')
+        except Exception as error:
+            logging.error('Source JSON file {} could not be loaded! '
+                            .format(filename))
+            logging.error('Error Message: {}'.format(error))
+            exit(0)
+    return data
+
+
+def calculate_additional_fields(data):
+    """
+    calculate_additional_fields
+    :param data:
+    :return:
+    """
+    logging.debug('Calculating additional data fields')
+    logging.debug('Inside calculate_additional_fields function')
+
+    for key, value in data.items():
+
+        logging.debug('Processing Rental Record: {}'.format(key))
+
+        try:
+            rental_start = datetime.datetime.strptime(
+                value['rental_start'], '%m/%d/%y')
+
+            rental_end = datetime.datetime.strptime(
+                value['rental_end'], '%m/%d/%y')
+
+            # Add new calculated values to data
+            value['total_days'] = (rental_end - rental_start).days
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+
+            logging.debug('Total Days: {}'.format(value['total_days']))
+            logging.debug('Total Price: {}'.format(value['total_price']))
+            logging.debug('Total Square Root of Price: {:.2f}'
+                          .format(value['sqrt_total_price']))
+            logging.debug('Total Unit Costs: {:.2f}'.format(value['unit_cost']))
+
+        except ValueError:
+            logging.warning('Error Processing Rental Record: {}.'.format(key))
+            logging.warning('Rental End Date {} before Start Date {}!'
```

Comment:
Are you sure that your code would return the message logging.warning('Rental End Date {} before Start Date {}!' ONLY in case when rental end data is before the start date...